### ISSUE TITLE
Auto properties section explains property initializers by linking to array initializers

### ISSUE THREAD

**Issue by jnm2** [dotnet/csharpstandard#1163]
<https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#1574-automatically-implemented-properties> links to the array initializers section to explain property initializers. Since not all properties are array properties, I would have expected it to link instead to <https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#1556-variable-initializers> or something else.

https://github.com/dotnet/csharpstandard/blob/02d1a90122ece425cff575f47f197fb396be73d4/standard/classes.md#L3557

**Comment by RexJaeschke** [comment]
[Although V7 is cited here, the same text exists in draft-v8.]

@jnm2 The sentence you cite contains "..., which is applied directly to the backing field as a *variable_initializer* ([§17.7](arrays.md#177-array-initializers))." The grammar rule *variable_initializer* is indeed defined in §17.7, so that reference is correct. It should *not* be implied that automatically implemented properties are explained anywhere in 17.7, just that is where that rule is defined. So, I don't see that any change is needed.

**Comment by jnm2** [comment]
Okay, thanks for confirming, I was unsure of the intentionality of it.

From a usability standpoint, it seems confusing for `variable_initializer` to be defined inside an arrays section (Arrays.md, "17.7 Array initializers".) Since variables are not necessarily of an array type, the "Array initializers" section or even the file are not places I would think to look to understand variable initializers. Why is it that `variable_initializer` is defined there, and does it sound realistic to consider giving variable initializers their own section outside the arrays section?

**Comment by RexJaeschke** [comment]
@jnm2 One might reasonably expect the grammar rule *variable_initializer* to be mentioned in the Variables chapter, but it is not. Also, rule *variable_declarator* is defined under Fields, and refers to *variable_initializer* (from Arrays).

In a large and complex spec, and one that has seen quite a bit of churn in the past 25 years, things get added to places that years later might not seem to be the "best" places, but for which there is little incentive to change.

Consider that most of the semantics of struct and interface members resides in the Classes chapter, which might not be where a new reader would expect. With the upcoming support for implementations in interfaces (V8), there have been discussions about moving the member stuff into its own chapter and having the others point there. However, that doesn't address various other issues.

When support for various kinds of patterns was added, we started out defining that topic locally in each affected section, but eventually spun all pattern-related stuff off into a new chapter. That was a big decision, and reorganizing the spec to handle change like that requires that we don't inadvertently break anything.

Bottom line, more than a few things could be done differently if we were starting from scratch. But, of course, we are not.

**Comment by jnm2** [comment]
That was a great explanation, thank you!
